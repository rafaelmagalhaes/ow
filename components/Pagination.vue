<template>
  <nav aria-label="Page navigation example">
    <ul class="inline-flex -space-x-px">
      <li>
        <button
          id="previous"
          :disabled="isFirstPage"
          :class="{ 'disabled:opacity-25': isFirstPage, 'cursor-pointer': !isFirstPage }"
          class="px-3 py-2 ml-0 leading-tight text-gray-500 bg-white border border-gray-300 rounded-l-lg hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
          @click="$emit('prev')"
        >
          Previous
        </button>
      </li>
      <li v-for="page in pages" :key="page">
        <button
          aria-current="page"
          :class="{
            'dark:bg-gray-700 dark:text-white text-blue-600': page === currentPage,
            'dark:hover:bg-gray-700 dark:bg-gray-800 dark:hover:text-white leading-tight':
              page !== currentPage,
          }"
          class="px-3 py-2 cursor-pointer text-gray-500 bg-white border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:border-gray-700 dark:text-gray-400"
          @click="$emit('onPageChange', page)"
        >
          {{ page }}
        </button>
      </li>

      <li>
        <button
          id="next"
          :disabled="isLastPage"
          :class="{ 'disabled:opacity-25': isLastPage, 'cursor-pointer': !isLastPage }"
          class="px-3 py-2 leading-tight text-gray-500 bg-white border border-gray-300 rounded-r-lg hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
          @click="$emit('next')"
        >
          Next
        </button>
      </li>
    </ul>
  </nav>
</template>
<script>
export default {
  name: 'Pagination',
  props: {
    totalPages: {
      type: Number,
      required: true,
    },
    isFirstPage: {
      type: Boolean,
      required: true,
    },
    isLastPage: {
      type: Boolean,
      required: true,
    },
    currentPage: {
      type: Number,
      required: true,
    },
  },
  setup(props) {
    const pages = computed(() => {
      const pa = [];
      for (let i = 1; i <= props.totalPages; i++) {
        pa.push(i);
      }
      return pa;
    });
    return { pages };
  },
};
</script>
