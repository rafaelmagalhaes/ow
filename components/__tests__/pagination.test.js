import { shallowMount } from '@vue/test-utils';
import { describe, it, expect, beforeEach } from 'vitest';
import Pagination from '~/components/Pagination.vue';

describe('Pagination', () => {
  let wrapper;
  beforeEach(async () => {
    wrapper = await shallowMount(Pagination, {
      propsData: {
        totalPages: 5,
        isFirstPage: false,
        isLastPage: false,
        currentPage: 3,
      },
    });
  });
  it('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  it('emits "prev" event when previous button is clicked', async () => {
    const button = wrapper.find('#previous');
    await button.trigger('click');
    expect(wrapper.emitted('prev')).toBeTruthy();
  });

  it('emits "next" event when next button is clicked', async () => {
    const button = wrapper.find('#next');
    await button.trigger('click');
    expect(wrapper.emitted('next')).toBeTruthy();
  });
  it('renders the correct number of page buttons, including next and prev buttons', () => {
    const buttons = wrapper.findAll('button');
    expect(buttons.length).toBe(7);
  });
});
