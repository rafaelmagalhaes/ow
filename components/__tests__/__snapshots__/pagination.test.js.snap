// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagination > renders correctly 1`] = `
<nav
  aria-label="Page navigation example"
>
  <ul
    class="inline-flex -space-x-px"
  >
    <li>
      <button
        class="cursor-pointer px-3 py-2 ml-0 leading-tight text-gray-500 bg-white border border-gray-300 rounded-l-lg hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
        id="previous"
      >
         Previous 
      </button>
    </li>
    
    <li>
      <button
        aria-current="page"
        class="dark:hover:bg-gray-700 dark:bg-gray-800 dark:hover:text-white leading-tight px-3 py-2 cursor-pointer text-gray-500 bg-white border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:border-gray-700 dark:text-gray-400"
      >
        1
      </button>
    </li>
    <li>
      <button
        aria-current="page"
        class="dark:hover:bg-gray-700 dark:bg-gray-800 dark:hover:text-white leading-tight px-3 py-2 cursor-pointer text-gray-500 bg-white border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:border-gray-700 dark:text-gray-400"
      >
        2
      </button>
    </li>
    <li>
      <button
        aria-current="page"
        class="dark:bg-gray-700 dark:text-white text-blue-600 px-3 py-2 cursor-pointer text-gray-500 bg-white border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:border-gray-700 dark:text-gray-400"
      >
        3
      </button>
    </li>
    <li>
      <button
        aria-current="page"
        class="dark:hover:bg-gray-700 dark:bg-gray-800 dark:hover:text-white leading-tight px-3 py-2 cursor-pointer text-gray-500 bg-white border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:border-gray-700 dark:text-gray-400"
      >
        4
      </button>
    </li>
    <li>
      <button
        aria-current="page"
        class="dark:hover:bg-gray-700 dark:bg-gray-800 dark:hover:text-white leading-tight px-3 py-2 cursor-pointer text-gray-500 bg-white border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:border-gray-700 dark:text-gray-400"
      >
        5
      </button>
    </li>
    
    <li>
      <button
        class="cursor-pointer px-3 py-2 leading-tight text-gray-500 bg-white border border-gray-300 rounded-r-lg hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
        id="next"
      >
         Next 
      </button>
    </li>
  </ul>
</nav>
`;
